src/fetcher/url_handler.js

```diff
@@ -66,8 +66,10 @@ async function get(url, options) {
       ...options,
       signal: controller.signal,
       credentials: options.withCredentials ? 'include' : 'omit',
+    })
+    .finally(()=>{
+      clearTimeout(timer);
```

Comment:
Can this be done in a `finally` block after the `catch` ? since `.finally()` is usually used in a Promise chain

Reply:
the `await` isn't that different that a promise chain, only different syntax. if you would want to move it to the try..catch..finally. the `const timer` needs to be moved outside the try block since it's block scoped. probably all together less neat.